mobile-app/app/screens/AppNavigator/screens/Portfolio/components/ActionButtons.tsx

```diff
@@ -0,0 +1,119 @@
+import { View } from '@components'
+import { IconName, IconType, ThemedIcon, ThemedTextV2, ThemedTouchableOpacityV2 } from '@components/themed'
+import { useFeatureFlagContext } from '@contexts/FeatureFlagContext'
+import { NavigationProp, useNavigation } from '@react-navigation/native'
+import { RootState } from '@store'
+import { futureSwapSelector } from '@store/futureSwap'
+import { tailwind } from '@tailwind'
+import { translate } from '@translations'
+import { ScrollView, Text } from 'react-native'
+import { useSelector } from 'react-redux'
+import { PortfolioParamList } from '../PortfolioNavigator'
+
+export interface ActionButtonsProps {
+  name: string
+  icon: IconName
+  iconType: IconType
+  iconSize?: number
+  onPress: () => void
+  testID: string
+  badge?: string | number
+}
+
+export function ActionButtons (): JSX.Element {
+  const { isFeatureAvailable } = useFeatureFlagContext()
+  const navigation = useNavigation<NavigationProp<PortfolioParamList>>()
+  const futureSwaps = useSelector((state: RootState) => futureSwapSelector(state))
+
+  const actions: ActionButtonsProps[] = [
+    {
+      name: translate('components/ActionButtons', 'Send'),
+      icon: 'arrow-up-right',
+      iconType: 'Feather',
+      iconSize: 28,
+      testID: 'send_balance_button',
+      onPress: () => navigation.navigate('Send')
+    }, {
+      name: translate('components/ActionButtons', 'Receive'),
+      icon: 'arrow-down-left',
+      iconType: 'Feather',
+      iconSize: 28,
+      testID: 'receive_balance_button',
+      onPress: () => navigation.navigate('Receive')
+    }, {
+      name: translate('components/ActionButtons', 'Swap'),
+      icon: 'repeat',
+      iconType: 'Feather',
+      testID: 'swap_tokens_button',
+      onPress: () => navigation.navigate({
+        name: 'CompositeSwap',
+        params: {},
+        merge: true
+      })
+    }, {
+      name: translate('components/ActionButtons', 'Transactions'),
+      icon: 'calendar',
+      testID: 'transaction_button',
+      iconType: 'Feather',
+      onPress: () => navigation.navigate('TransactionsScreen')
+    }
+  ]
+
+  if (isFeatureAvailable('future_swap') && futureSwaps.length > 0) {
```

Comment:
@fullstackninja864 I feel this one is more expensive. What if we just use if statements for rendering?